include/oc_push.h

```diff
@@ -0,0 +1,232 @@
+/****************************************************************************
+ *
+ * Copyright 2021 ETRI All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * Created on: Aug 23, 2021,
+ * 				Author: Joo-Chul Kevin Lee (rune@etri.re.kr)
+ *
+ *
+ ****************************************************************************/
+
+#ifndef OC_PUSH_H
+#define OC_PUSH_H
+
+#include <stdio.h>
+
+#include "oc_config.h"
+#include "oc_helpers.h"
+#include "oc_rep.h"
+#include "oc_ri.h"
+#include "oc_endpoint.h"
+#include "port/oc_log.h"
+#include "util/oc_memb.h"
+#include "util/oc_process.h"
+
+
+#if defined(OC_PUSHDEBUG) || defined(OC_DEBUG)
+#ifndef OC_DEBUG
+#define oc_print(...) printf(__VA_ARGS__)
+#define oc_log(level, ...)                                                     \
```

Comment:
This is the same macro as in `port/oc_log.h`, right? In that case would reuse the `OC_LOG` define from there and modify the conditions there so it is defined for `OC_PUSHDEBUG`:

Changes to `port/oc_log.h`
```
#if defined(OC_DEBUG) || defined(OC_PUSHDEBUG)
#ifdef __ANDROID__
#define OC_LOG(level, ...)                                                     \
  android_log(level, __FILE__, __func__, __LINE__, __VA_ARGS__)
...
#endif /* __ANDROID__ */
#endif /* OC_DEBUG || OC_PUSHDEBUG */

#ifdef OC_DEBUG
#define OC_DBG(...) OC_LOG("D", __VA_ARGS__)
#define OC_WRN(...) OC_LOG("W", __VA_ARGS__)
#define OC_ERR(...) OC_LOG("E", __VA_ARGS__)
```

So now the `OC_LOG` would be defined whenever at least one of `OC_DEBUG` or `OC_PUSHDEBUG` is defined. `OC_LOG` can be used in `oc_push.h` and `OC_DBG`/`OC_WRN`/`OC_ERR` are not defined if `OC_DEBUG` is not defined.

Reply:
I agree, fixed !